### DISCUSSION TITLE
How to change the owner of a file?

### DISCUSSION THREAD

**Discussion by adanielyan** [directus/directus#16749] [Archive]
I don't see a way to change the owner (uploaded_by) of the file. This field is locked in directus_files collection. I believe admin user should be able to change the owner of the file.

**Comment by br41nslug** [comment]
You may need to use the API to update that value https://docs.directus.io/reference/files.html#update-a-file

**Reply by adanielyan** [reply]
Yeah, I considered that option, but having to use API every time I need to change an ownership of a file is suboptimal. I use the ownership for permission control, and sometimes admin needs to assign files to other users, even if the file was uploaded by admin.

Since there is already a UI for changing ```modified_by``` field, I think having a UI for changing the ```uploaded_by``` would be the most appropriate.

As a workaround I am thinking to add another relationship field ```User``` on files collection and set it to default value of a current user with ability to change it by admin.

**Reply by rijkvanzanten** [reply]
> As a workaround I am thinking to add another relationship field User on files collection and set it to default value of a current user with ability to change it by admin.

That's what I'd recommend doing! `uploaded_by` is not the same as `owner` semantically speaking, so having a separate field for that makes sense to me 🙂

**Reply by adanielyan** [reply]
Thank you for asserting my approach!

Just out of curiosity, what is the rationale of having a user interface for changing the `modified_by` field, but not for `uploaded_by`?

**Reply by rijkvanzanten** [reply]
> Just out of curiosity, what is the rationale of having a user interface for changing the modified_by field, but not for uploaded_by?

Neither of those are rendered on the file detail page afaik 🤔  Both of those are readonly accountability fields so you can keep track of origin of files

**Reply by adanielyan** [reply]
I just checked, and you are right, I cannot change either. But it must have been changed in one of the recent versions as I am pretty sure I was able to modify it in the past  😳
